### ISSUE TITLE
False-positive nullability warning

### ISSUE THREAD

**Issue by DoctorKrolic** [dotnet/roslyn#64117]
**Version Used**:
https://github.com/dotnet/roslyn/commit/60e74b70fab1122e7196c2d9b033677a39e9b76d

**Steps to Reproduce**:
```cs
#nullable enable

void M(MyClass? myClass)
{
    var guardedProp = myClass?.Prop;

    if (guardedProp is null)
    {
        return; // or throw
    }

    var prop = myClass.Prop; // CS8602 on `myClass`
}

class MyClass
{
    public string Prop { get; set; } = null!;
}
```

**Expected Behavior**:
No `CS8602` where I marked because the value cannot be null at that point

**Actual Behavior**:
`CS8602: Dereference of a possibly null reference.`

**Comment by CyrusNajmabadi** [comment]
Roslyn doesn't track the relation between separate variables. This behavior is expected. @RikkiGibson in case this is a dupe that he can link.

**Comment by DoctorKrolic** [comment]
And there is no planning for supporting it?

**Comment by RikkiGibson** [comment]
marking *not planned*, i.e. *won't fix/by design* makes sense to me.

![image](https://user-images.githubusercontent.com/5833655/191123970-e45dd21b-2501-44c1-a0f4-395502142593.png)

**Comment by DoctorKrolic** [comment]
Thank you @RikkiGibson! I might be a bit annoying asking to reclose issues, but I just wanna more devs to know about this cool github feature and use it more often
